### ISSUE TITLE
Inconsistent behaviour of selectors when n_select_features > rank(all_features)

### ISSUE THREAD

**Issue by agoscinski** [cosmo-epfl/scikit-cosmo#60]
CUR should not throw an error if n_select_features > rank(all_features)

```py
import numpy as np
from skcosmo.selection import CUR

X = np.ones((10,2))

n_features = 1
fs = CUR.FeatureCUR(X)
fs.select(n_features)
print("No error because rank of X is 1")

# Error because rank of X is < 2
n_features = 2
fs = CUR.FeatureCUR(X)
fs.select(n_features)
```
with error output
```
  [...]
  File "/home/goscinsk/miniconda3/lib/python3.8/site-packages/skcosmo/utils/orthogonalizers.py", line 29, in X_orthogonalizer
    raise ValueError("Cannot orthogonalize by a null vector.")
ValueError: Cannot orthogonalize by a null vector.
```

I would be fine for my usage cases with CUR outputting a warning and adding random features. Also an additional option to stop feature selection in this case would be nice.

**Comment by agoscinski** [comment]
Similar problem with FPS.
```py
import numpy as np
from skcosmo.selection import FPS

X = np.ones((10,3))

n_features = 1
fs = FPS.FeatureFPS(X)
fs.select(n_features)
print("X.shape", X.shape)
print("n_features", n_features)
print("len(fs.idx)", len(fs.idx))
print("len(np.unique(fs.idx))", len(np.unique(fs.idx)))
print()

n_features = 3
fs = FPS.FeatureFPS(X)
fs.select(n_features)
print("X.shape", X.shape)
print("n_features", n_features)
print("len(fs.idx)", len(fs.idx))
print("len(np.unique(fs.idx))", len(np.unique(fs.idx)))
```
Output
```
X.shape (10, 3)
n_features 1
len(fs.idx) 1
len(np.unique(fs.idx)) 1

X.shape (10, 3)
n_features 3
len(fs.idx) 2
len(np.unique(fs.idx)) 2
```
FPS does only select features until the distance is close to zero and then stops. This could be FPS default behaviour, but it is really not clear from the python doc and the naming of the variables, and not consistent with CUR at the moment. I would rather have "select me exactly this amount, does not matter if these are crappy features" as default behaviour since PCA also does it this way and gives me an option to select the number of features in a smart way.

I also had the issue that one index was two times in `fs.idx`, but this was in an older version of skcosmo, so could have already been fixed

**Comment by agoscinski** [comment]
Also there is somewhere an idx check wrong for FPS such that one index is included twice. I haven been able to reproduce this for a small script, but here a snippet of a larger part of the code

```py
        fs = skcosmo.selection.FPS.FeatureFPS(features, tol=1e-50)
        fs.select(self.n_features)
        self.selected_idx_ = fs.idx
        print("len(self.selected_idx_)",len(self.selected_idx_))
        print("len(np.unique(self.selected_idx_))",len(np.unique(self.selected_idx_)))
```
Output
```
len(self.selected_idx_) 391
len(np.unique(self.selected_idx_)) 390
```

**Comment by rosecers** [comment]
Addressed in #58
